src/bun.js/bindings/bindings.cpp

```diff
@@ -616,6 +619,24 @@ bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2,
     case JSFunctionType: {
         return false;
     }
+
+    case JSDOMWrapperType: {
+        if (c2Type == JSDOMWrapperType) {
+            // https://github.com/oven-sh/bun/issues/4089
+            if (auto* url1 = jsDynamicCast<JSDOMURL*>(v1)) {
+
+                if (auto* url2 = jsDynamicCast<JSDOMURL*>(v2)) {
+                    return url1->wrapped().href() == url2->wrapped().href();
+                }
+
+                if constexpr (isStrict) {
+                    return false;
+                }
+            }
+        }
```

Comment:
should we return false if one is a url and the other isn't?

Reply:
maybe just for isStrictEqual?

Reply:
yeah, that sounds good